File: Reqnroll/Chapter18/18.5-SkippableResetPattern/After/WIMP.Specs/Drivers/StubDatabaseDriver.cs
namespace WIMP.Specs.Drivers;

public class StubDatabaseDriver : IDatabaseDriver
{
    public Task EmptyDatabase(IReadOnlyCollection<string>? exceptTables = null)
    {
        //nop
        return Task.CompletedTask;
    }

    public Task UpgradeSchemaIfNeeded()
    {
        //nop
        return Task.CompletedTask;
    }

    public Task<bool> WasTableModified(string tableName)
    {
        return Task.FromResult(true); // treat it modified, seeding is anyway "free"
    }

    public Task ResetTableModificationTracking()
    {
        //nop
        return Task.CompletedTask;
    }
}

File: CucumberJVM/Chapter18/18.5-SkippableResetPattern/After/src/test/java/com/wimp/app/specs/drivers/StubDatabaseDriver.java
package com.wimp.app.specs.drivers;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

import java.util.Collection;

@Component
@ConditionalOnProperty(name = "test.database.use-stub", havingValue = "true")
public class StubDatabaseDriver implements DatabaseDriver {

    @Override
    public void emptyDatabase(Collection<String> exceptTables) {
        // nop
    }

    @Override
    public boolean wasTableModified(String tableName) {
        return true; // treat it modified, seeding is anyway "free"
    }

    @Override
    public void resetTableModificationTracking() {
        // nop
    }
}
